internal/pkg/describe/lb_web_service.go

```diff
@@ -66,8 +68,8 @@ func (s *serviceDiscovery) String() string {
 }
 
 type svcDescriber interface {
-	Params() (map[string]string, error)
-	EnvOutputs() (map[string]string, error)
+	EnvStackParamOutput() (params map[string]string, outputs map[string]string, err error)
+	StackParamOutput() (params map[string]string, outputs map[string]string, err error)
```

Comment:
This interface feels like is due a refactor. There are too many awkward things 😅   

For example, `StackParamOutput` returns stack parameters and outputs but then we have a separate method for just returning the resources. It's not clear why it's separate.  
The interface named `svcDescriber` also fetches outputs from the env stack which is peculiar 😆 

Can we find a better separation? What about this:
```go
type cfnStackDescriber interface {
   Params() (map[string]string, error)
   Outputs() (map[string]string, error)
   StackResources() ([]*cloudformation.StackResource, error)
}

type ecsSvcDescriber interface {
   cfnStackDescriber
   EnvVars() ([]*ecs.ContainerEnvVar, error)
   Secrets() ([]*ecs.ContainerSecret, error)
}

// implements cfnStackDescriber
type envStackDescriber struct { ... }

// implements ecsSvcDescriber
type ecsSvcDescriber struct { ... }
```
Anyway it doesn't need to be like the above, but it would be great if we can refactor this interface a little.

Reply:
Addressed!